tests/integration/setup_environment.py

```diff
@@ -0,0 +1,89 @@
+from typing import Self
+
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.network import Network
+from testcontainers.core.waiting_utils import wait_container_is_ready, wait_for_logs
+from testcontainers.postgres import PostgresContainer
+
+postgres_image_name = "postgres:latest"
+adcm_image_name = "hub.adsw.io/adcm/adcm:develop"
+adcm_port = 8000
+postgres_port = 5432
+adcm_container_name = "test_adcm"
+postgres_name = "test_pg_db"
+db_user = "adcm"
+db_name = "adcm"
+db_password = "password"  # noqa: S105
+
+
+class ADCMPostgresContainer(PostgresContainer):
+    url: str
+
+    def __init__(self: Self, image: str) -> None:
+        super().__init__(image)
+
+    def setup_postgres(self: Self, username: str, password: str, adcm_db_name: str, container_network: Network) -> None:
+        self.with_name(postgres_name)
+        self.with_bind_ports(postgres_port, postgres_port)
+        self.password = password
+        self.with_network(container_network)
+
+        self.start()
+        wait_container_is_ready(self)
+
+        init_db_cmds = (
+            f"CREATE USER {username} WITH ENCRYPTED PASSWORD '{password}'; "
+            f"CREATE DATABASE {adcm_db_name} OWNER {username};"
+        )
+        self.with_command(
+            f"PGPASSWORD={self.password} psql --username {self.username} --host 127.0.0.1 -c '{init_db_cmds}'"
```

Comment:
looks like there will be a problem with `'` in `'` (in `init_db_cmds` and within `with_command`)

_maybe_ it's better to send queries directly to PostgreSQL

Reply:
would need to bring something like sqlalchemy just for this one command then. Maybe better to avoid it?

Reply:
no need in sqlalchemy

looks like working for now, so np